### PR TITLE
Plumb a reference of PlatformViewsController and AccessibilityBridge to each other

### PR DESCRIPTION
**Pull request by amirh**
This is in preparation for implementing platform views a11y on Android.

And e2e working prototype is available here: https://github.com/amirh/engine/tree/a11y_hacks

https://github.com/flutter/flutter/issues/19418

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java [hunk 1]
```diff
@@ -443,12 +444,15 @@ public void attachToFlutterEngine(@NonNull FlutterEngine flutterEngine) {
...
+    PlatformViewsController platformViewsController = flutterEngine.getPluginRegistry().getPlatformViewsController();
     accessibilityBridge = new AccessibilityBridge(
         this,
         flutterEngine.getAccessibilityChannel(),
         (AccessibilityManager) getContext().getSystemService(Context.ACCESSIBILITY_SERVICE),
-        getContext().getContentResolver()
+        getContext().getContentResolver(),
+        platformViewsController
     );
+    platformViewsController.attachAccessibilityBridge(accessibilityBridge);
```
Comment (goderbauer) [inline shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java]: Do we need to detach this ever?

Comment (amirh) [reply shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java]: I tried to keep the same lifespan as the AccessibilityBridge reference, should `accessibilityBridge` be unset in the new FlutterView like in the current one?

Comment (goderbauer) [reply shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java]: @matthew-carroll May have more context to answer that question.

**Comment by matthew-carroll** [comment]
@amirh can you avoid the new FlutterView and just make this change to the old FlutterView for now?

Also, I don't think we should have cyclical dependencies like this between `AccessibilityBridge` and `PlatformViewsController`.  This indicates a deeper problem to me.  One can use the other, but if they both require each other to exist all the time, then I think that's an indication of a likely object decomposition problem.

**Comment by amirh** [comment]
@matthew-carroll sure I'll remove it from the in progress FlutterView, I'm leaving a TODO for you to do the plumbing there.

I agree this cyclic dependency is smelly, but I'm not sure there's much we can do other than merge the whole thing into a single object: as we are proxying a 2-way communication channel between the Android a11y framework and the view.
Note that they do not "require each other to exist all the time", the platform views controller can live without a reference to the a11y bridge and in this case will just not propagate a11y events out of the virtual display.

**Comment by matthew-carroll** [comment]
I'm confused as to what the relationship is.  I don't see any usages of either artifact by the other, just references held.  I think we need to see what the actual behavior is if we're going to evaluate the necessity of this relationship.  I could imagine a few alternatives, but they depend entirely on the specific interactions.

**Comment by amirh** [comment]
You can check the (hacky) prototype to get an idea of the interactions, that change is going to be hard to review at once so I'm leading incremental pieces, with this being the first step.

**Comment by matthew-carroll** [comment]
I think it's pretty important that we review the actual behaviors before giving an LGTM to the fundamental relationship.  Can you describe why it will be difficult to review at one time?  If you'd like, I can come by and we can chat in person if that would make it easier.  But I'm really not comfortable with this dependency in the absence of the supporting purpose for its existence.

**Comment by amirh** [comment]
@goderbauer PTAL

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsAccessibilityDelegate.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.platform;
+
+/**
+ * Facilitates interaction between the accessibility bridge and embedded platform views.
+ */
+public interface PlatformViewsAccessibilityDelegate {
+}
```
Comment (goderbauer) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewsAccessibilityDelegate.java]: Should there be a TODO in here to fill this with ... stuff (I am wondering what that would be?)?

Comment (amirh) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsAccessibilityDelegate.java]: Added a TODO

**Comment by matthew-carroll** [comment]
Amir and I spoke about the relationship between `AccessibilityBridge` and the platform views system.  We agreed on an approach in which `AccessibilityBridge` will hang a dependency on what Amir has called the `PlatformViewsDelegate`, and that delegate will offer an interface capable of returning a `View` to `AccessibilityBridge`.  The `AccessibilityBridge` will then be updated to extract accessibility information out of that `View` and send it to Android, as well as forward accessibility events to the given `View`.

LGTM

**Pull request merged**